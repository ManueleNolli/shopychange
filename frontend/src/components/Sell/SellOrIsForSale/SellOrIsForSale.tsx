import React from 'react'
import { NFT } from '../../../types/components/NFT'
import { Sale } from '../../../types/components/Sale'
import { SaleStatus } from '../../../types/components/SaleStatus'
import { colors } from '../../../styles/Colors'
import ShadowButton from '../../ShadowButton/ShadowButton'
import SellManager from '../SellManager/SellManager'

type SellOrIsForSaleProps = {
  nft: NFT
  onUpdate: () => void
  sale: Sale | null
}

export default function SellOrIsForSale({
  nft,
  onUpdate,
  sale,
}: SellOrIsForSaleProps) {
  if (sale?.status !== SaleStatus.LISTED) {
    return <SellManager nft={nft} onUpdate={onUpdate} />
  } else {
    return (
      <ShadowButton
        isDisabled={true}
        label={'For Sale'}
        bgColor={`${colors.forSaleChakraColor}.400`}
        textColor={'white'}
        shadowColor={colors.forSaleShadowColor}
        hoverColor={`${colors.forSaleChakraColor}.500`}
        focusColor={`${colors.forSaleChakraColor}.500`}
      />
    )
  }
}
