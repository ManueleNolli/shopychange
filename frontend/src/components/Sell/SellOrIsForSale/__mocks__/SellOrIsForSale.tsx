import { divWithoutChildrenMock } from '../../../../../mocksUtils'

const SellOrIsForSaleMock = () => divWithoutChildrenMock('SellOrIsForSale')
export default SellOrIsForSaleMock
