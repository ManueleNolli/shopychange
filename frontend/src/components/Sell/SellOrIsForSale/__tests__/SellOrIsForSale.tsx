import React from 'react'
import { render } from '@testing-library/react'
import SellOrIsForSale from '../SellOrIsForSale'

const onUpdate = jest.fn()
const mockNFT = {
  contractAddress: 'mockContractAddress',
  tokenId: 0,
  image: 'mockImage',
  name: 'mockName',
}

const mockSale = {
  nftData: {
    contractAddress: 'mockContractAddress',
    tokenId: 0,
    image: 'mockImage',
    name: 'mockName',
  },
  seller: 'mockSeller',
  price: 0,
  status: 0,
}

jest.mock('../../SellManager/SellManager')

describe('SellOrIsForSale', () => {
  test('renders correctly if not listed', () => {
    const tree = render(
      <SellOrIsForSale nft={mockNFT} sale={mockSale} onUpdate={onUpdate} />
    )
    expect(tree.container).toMatchSnapshot()
  })
  test('renders correctly if listed', () => {
    const mockSaleWithStatus = {
      ...mockSale,
      status: 3,
    }

    const tree = render(
      <SellOrIsForSale
        nft={mockNFT}
        sale={mockSaleWithStatus}
        onUpdate={onUpdate}
      />
    )
    expect(tree.container).toMatchSnapshot()
  })
})
