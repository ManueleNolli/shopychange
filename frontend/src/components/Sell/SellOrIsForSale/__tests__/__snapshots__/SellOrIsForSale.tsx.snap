// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellOrIsForSale renders correctly if listed 1`] = `
<div>
  <button
    class="chakra-button css-1l55t55"
    data-testid="onClick-button"
    disabled=""
    style="white-space: initial;"
    type="button"
  >
    For Sale
  </button>
</div>
`;

exports[`SellOrIsForSale renders correctly if not listed 1`] = `
<div>
  <div
    data-testid="SellManager"
  />
</div>
`;
